hbw/tasks/dy_corrections.py

```diff
@@ -0,0 +1,484 @@
+        denominator = h["MC_corr_process"][{ptll_var: sum}].values()
+        njet_rate = safe_div(nominator, denominator)
+
+        return njet_rate
+
+    # build fit function string
+    def get_fit_str(njet: int, njet_overflow: int, rate_factor: float, h: hist.Hist) -> dict:
+        from scipy import optimize
+
+        # Check if number of jets are valid
+        total_n_jet = np.arange(1, 12)
+        if njet not in total_n_jet:
+            raise ValueError(f"Invalid njets value {njet}, expected int between 1 and 11.")
+
+        # Helper function to plot the fit and string function
+        def plot_fit_func(fit_func, fit_string, param_fit, rate_factor, ratio_values, njet):
+            #  plot the fit function and the string function (as sanity)
+            def str_function(x, fit_str):
+                from scipy import special
+                fit_str = fit_str.replace("^2", "**2")
+                str_function = eval(fit_str, {"x": x, "exp": np.exp, "erf": special.erf})
+                return str_function
+
+            s = np.linspace(0, 400, 1000)
+            y = [(rate_factor * fit_func(v, *param_fit)) for v in s]
+            z = [str_function(v, fit_string) for v in s]
+            fig, ax = plt.subplots()
+            ax.plot(s, y, color="grey", label="Fit function")
+            ax.plot(s, z, color="black", linestyle="--", label="String function")
+            ax.errorbar(
+                bin_centers,
+                ratio_values,
+                yerr=ratio_err,
+                fmt=".",
+                color="black",
+                linestyle="none",
+                ecolor="black",
+                elinewidth=0.5,
+            )
+            ax.legend(loc="upper right")
+            ax.set_xscale("log")
+            ax.set_xlabel(r"$p_{T,ll}\;[\mathrm{GeV}]$")
+            ax.set_ylabel("Ratio")
+            ax.set_ylim(0.4, 1.5)
+            ax.set_title(f"Fit function for NLO 2022postEE, njets {njet}")
+            ax.grid(True)
+            fig.savefig(f"plot_fit_njet{njet}.pdf")
+
+        # Define binning
+        if njet < njet_overflow:
+            jet_tuple = (njet, njet + 1)
+        elif njet >= njet_overflow:
+            jet_tuple = (njet_overflow, 11)
+
+        data_h = h["data"][hist.loc(jet_tuple[0]):hist.loc(jet_tuple[1]), ...]
+        dy_h = h["MC_corr_process"][hist.loc(jet_tuple[0]):hist.loc(jet_tuple[1]), ...]
+        mc_h = h["MC_subtract"][hist.loc(jet_tuple[0]):hist.loc(jet_tuple[1]), ...]
+
+        # get bin centers
+        bin_centers = dy_h.axes[-1].centers
+
+        # get histogram values and errors, summing over njets axis
+        dy_values = (dy_h.view().value).sum(axis=0)
+        data_values = (data_h.view().value).sum(axis=0)
+        mc_values = (mc_h.view().value).sum(axis=0)
+        dy_err = (dy_h.view().variance**0.5).sum(axis=0)
+        data_err = (data_h.view().variance**0.5).sum(axis=0)
+        mc_err = (mc_h.view().variance**0.5).sum(axis=0)
```

Comment:
Sum first, then sqrt. Or directly pass the variances to calculate the `ratio_err`

```suggestion
        dy_err = (dy_h.variances().sum(axis=0)**0.5
        data_err = (data_h.variances().sum(axis=0)**0.5
        mc_err = (mc_h.variances().sum(axis=0)**0.5
```